source/core_http_client.c

```diff
@@ -926,20 +950,34 @@ static int httpParserOnMessageCompleteCallback( http_parser * pHttpParser )
 
 /*-----------------------------------------------------------*/
 
-static void initializeParsingContextForFirstResponse( HTTPParsingContext_t * pParsingContext )
+static void initializeParsingContextForFirstResponse( HTTPParsingContext_t * pParsingContext,
+                                                      const HTTPRequestHeaders_t * pRequestHeaders )
 {
     assert( pParsingContext != NULL );
+    assert( pRequestHeaders->headersLen >= HTTP_MINIMUM_REQUEST_LINE_LENGTH );
```

Comment:
Do we need an assert to make sure pRequestHeaders is not NULL?

Reply:
Yes thanks